### DISCUSSION TITLE
Storefront menu is empty after Swell major update

### DISCUSSION THREAD

**Discussion by agsmartwolf** [swellstores/community#200] [Help] [answered]
It was a menu item "navigation" in "Storfront" section on the dashboard before.
I had menu settings for site navigation for my custom storefront there.

The menuSections field in the settings response is empty, so my store can't be built. The store's core is based on the `horizon` storefront. I suppose "horizon" is broken too.

<img width="376" alt="image" src="https://user-images.githubusercontent.com/109105178/230416542-7c612bd2-5022-4db1-8ceb-730a60027bfb.png">

<img width="1354" alt="Screenshot 2023-04-06 at 18 52 06" src="https://user-images.githubusercontent.com/109105178/230416985-ab73de2d-695c-4317-b664-3249803d517e.png">

**Comment by agsmartwolf** [comment]
`storeData.data.storeSettings.values` - is empty.
<img width="1354" alt="Screenshot 2023-04-06 at 19 44 34" src="https://user-images.githubusercontent.com/109105178/230430319-915d7999-42b5-4a53-88a8-2aa9d21029ca.png">

**Comment by logeshswell** [comment, marked as answer]
@agsmartwolf I updated your menus and store settings now. Can you check and confirm once?

**Reply by agsmartwolf** [reply]
Yes, it works. Thank you!

**Reply by pageperfectco** [reply]
@logeshswell we are having a similar issue. The menu options in the Storefront navigation is missing. Can you restore this for us? Our store is mountain-tactical-company

**Reply by logeshswell** [reply]
@pageperfectco I updated your menus and store settings now. Can you check and confirm once?

**Reply by pageperfectco** [reply]
@logeshswell thanks for trying. Unfortunately, we still don't have the Navigation menu for setting menu navigation.

**Reply by ericingram** [reply]
We're close to deploying a fix to replace the Navigation editor, likely tomorrow morning (US time)

**Comment by brahima** [comment]
Hello,

Seems that i'm hitting the same issue (see code below) with the horizon theme.
In fact i was actually developing early in the morning (6AM / Paris Time Zone) when the API outage occurred ... so i had to do other things.
Despite several hours have passed, i still have the same issue.
I tried creating another test store ... same error. 
My stores are (kotoboka-dev and kotoboka-test-1)

`TypeError: Cannot read properties of undefined (reading 'menu')
    at formatStoreSettings (webpack-internal:///./utils/settings.ts:72:39)
    at getStoreSettings (webpack-internal:///./lib/shop/fetchingFunctions.ts:323:79)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async eval (webpack-internal:///./lib/utils/fetch_decorators.ts:23:56)`

Any help would be greatly appreciated.

**Reply by logeshswell** [reply]
I updated the missing configurations to kotoboka-dev store. Please check and confirm.
kotoboka-test-1 store wasn't created completely and interrupted in the middle due to API outage. So, ignore this store and try creating another one if required.

**Reply by brahima** [reply]
Hello @logeshswell and thanks.
It's ok now. I also updated the other discussion.
Regards

**Reply by brahima** [reply]
@logeshswell Just to let you know that I've just tried to create another new store (kotoboka-test-2) but still facing the same issue regarding the empty settings (unable to build / run using thee horizon theme).

Can you confirm this on your side please and help me fix this.
Thanks in advance

**Reply by logeshswell** [reply]
@brahima There is an issue with a new storefront initialization in the recent release. Our engineers are working on it and I will let you know once it is fixed.

**Reply by brahima** [reply]
Hello @logeshswell and thanks for your feedback.
This one is really critical/blocking at this point since i can't make any progress and i'm really a few steps away from publishing my store.
Really looking forward to hearing from you :-)
Regards, Brahim

**Reply by logeshswell** [reply]
@brahima The latest version of Horizon now includes fixes to the issues mentioned in this thread. Please pull the latest changes and run `npm install`, you'll also have to create a `SWELL_STOREFRONT_ID` env var. More info can be found [here](https://developers.swell.is/storefronts/horizon).

**Comment by agsmartwolf** [comment]
What is the right way to edit the menu and translations for different pages and sections right now?
I've tried to use swell CLI tools but couldn't figure out how to fetch and upload translations.

(I'd like to have an opportunity to fetch and change products translations too - if it's possible)

I have the next case:

We have a custom storefront, based on the horizon theme. 
For the first step, we used a storefront dashboard for the **original** theme to set navigation, translations for pages, some sections and checkout translations. After that, we enabled a custom storefront and set a storefront URL.
Settings had disappeared after that step (as for now). If we would like to change some of them: we had turned off a custom storefront using a specific checkbox in storefront settings, changed translations or some settings, and turned on the custom storefront again.

**Reply by logeshswell** [reply]
> What is the right way to edit the menu and translations for different pages and sections right now?

That's a brilliant question, Moving forward, you could edit the navigation menus and translations via APIs using PUT request as follows:
* Menus - https://smart-wolf.swell.store/admin/console?get/%3Astorefronts/-UHQ49lwf~Hconfigs~Hmenus~Hvalues~Hmenus
* Translations - https://smart-wolf.swell.store/admin/console?get/%3Astorefronts/-UHQ49lwf~Hconfigs~Heditor~Hvalues

**Reply by logeshswell** [reply]
> I have the next case:

A recent update to our storefront dashboard removed some functionality connected to Swell-hosted storefronts to improve the experience of hosting storefronts directly with 3rd party providers, such as Vercel or Netlify.

With that, the navigation section which was part of the hosted storefront feature was removed in this recent update. But still, you could edit navigation, translations for products, pages, sections, and checkout inside the editor page as described in the [previous](https://github.com/orgs/swellstores/discussions/200#discussioncomment-5551766) thread.

**Reply by logeshswell** [reply]
On a side note, I like your storefront very much and I can see that you made the best use of our Swell system. :)

**Reply by ericingram** [reply]
Note we are also working to restore the UI for editing nav menus outside of the theme editor. Most likely will be deployed next week.

**Reply by TotallWAR** [reply]
> On a side note, I like your storefront very much and I can see that you made the best use of our Swell system. :)

Thank you for your warm words! 


And thanks for clarifying the menu and translation editing options.

**Reply by ericingram** [reply]
Beautiful site indeed! https://www.smart-wolf.com/

<img width="1777" alt="image" src="https://user-images.githubusercontent.com/952134/230650667-4cd17740-cba6-4b75-beaf-0d85408dc739.png">
